src/compiler/checker.ts

```diff
@@ -11040,8 +11040,7 @@ namespace ts {
 
         function hasContextSensitiveReturnExpression(node: FunctionLikeDeclaration) {
             // TODO(anhans): A block should be context-sensitive if it has a context-sensitive return value.
-            const body = node.body!;
-            return body.kind === SyntaxKind.Block ? false : isContextSensitive(body);
+            return !node.body || node.body.kind === SyntaxKind.Block ? false : isContextSensitive(node.body);

```

Comment:
I don't think a nice without a body should be marked as context sensitive TBH. Makes more sense as `node.body && (...rest...)`

Reply:
Oh, wait, priority. This and the other part of the `||` are in the condition of the ternary (I guess). NVM.

Reply:
why does this need to be a ternary in the first place? IMO that's pretty confusing.
Why not simply `return !!node.body && node.body.kind === SyntaxKind.Block && isContextSensitive(node.body);`?